hasktorch/src/Torch/NN.hs

```diff
@@ -247,30 +246,30 @@ instance Randomizable LinearSpec Linear where
         =<< pure
           ( subScalar bound $ mulScalar (bound * 2.0) init
           )
-
     return $ Linear w b
 
 --
 -- Conv2d
 --
 
-data Conv2dSpec = Conv2dSpec
-  { inputChannelSize :: Int,
-    outputChannelSize :: Int,
-    kernelHeight :: Int,
-    kernelWidth :: Int
-  }
+data Conv2dSpec
+  = Conv2dSpec
+      { inputChannelSize :: Int,
+        outputChannelSize :: Int,
+        kernelHeight :: Int,
+        kernelWidth :: Int
+      }
   deriving (Show, Eq)
 
-data Conv2d = Conv2d
-  { conv2dWeight :: Parameter,
-    conv2dBias :: Parameter
-  }
+data Conv2d
+  = Conv2d
+      { conv2dWeight :: Parameter,
+        conv2dBias :: Parameter
+      }
   deriving (Show, Generic, Parameterized)
 
 conv2dForward :: Conv2d -> (Int, Int) -> (Int, Int) -> Tensor -> Tensor
-conv2dForward layer stride padding input =
-  Torch.Functional.conv2d' w b stride padding input
+conv2dForward layer = Torch.Functional.conv2d' w b
```

Comment:
it would be better if this wasn't eta reduced

Reply:
Would it help if I document the arguments instead of undoing the eta-reduce? 😉

Reply:
It would, I was afraid to ask though. This could be a lot of work, since this has happened many times in this PR. It would be an awesome contribution, @kutyel! Thanks for offering 😊